src/zenml/orchestrators/step_runner.py

```diff
@@ -118,8 +130,17 @@ def run(
             step_failed = False
             try:
                 return_values = step_entrypoint(**function_params)
-            except:  # noqa: E722
+            except Exception as step_exception:  # noqa: E722
```

Comment:
```suggestion
            except BaseException as step_exception:  # noqa: E722
```

I just realized that we actually want this here instead (this is equivalent to the empty `except` and also catches KeyboardInterrupt etc. The hook function then also needs to accept this type I think